crates/starknet-server/src/api/models/transaction.rs

```diff
@@ -232,19 +230,27 @@ pub struct EventContent {
 pub struct EventFilter {
     pub from_block: Option<BlockId>,
     pub to_block: Option<BlockId>,
+    pub address: Option<ContractAddress>,
+    pub keys: Option<Vec<Vec<Felt>>>,
     pub continuation_token: Option<String>,
     pub chunk_size: usize,
-    pub address: Option<ContractAddress>,
-    #[serde(default)]
-    pub keys: Vec<HashSet<Felt>>,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct EventsChunk {
-    pub events: Vec<Event>,
+    pub events: Vec<EmittedEvent>,
     pub continuation_token: Option<String>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub struct EmittedEvent {
```

Comment:
It would be good to define new types in `types` crate. To save effort of refactoring

Reply:
I will add it

Reply:
https://github.com/0xSpaceShard/starknet-devnet-rs/commit/f54fb47b91f7a0761329c17406a6a4403cd0c7f0